src/protocol/requests/api-versions/api-versions-request-builder.ts

```diff
@@ -0,0 +1,78 @@
+import { TagSection } from '../../commons';
+import { type CompactString, Int16, Int32, type NullableString } from '../../primitives';
+import { RequestHeaderV1, RequestHeaderV2 } from '../headers';
+import { type RequestBuilder } from '../request-builder';
+import { ApiVersionsRequestV0 } from './api-versions-request-v0';
+import { ApiVersionsRequestV1 } from './api-versions-request-v1';
+import { ApiVersionsRequestV2 } from './api-versions-request-v2';
+import { ApiVersionsRequestV3 } from './api-versions-request-v3';
+
+export type ApiVersionsRequest =
+  | ApiVersionsRequestV0
+  | ApiVersionsRequestV1
+  | ApiVersionsRequestV2
+  | ApiVersionsRequestV3;
+
+export class ApiVersionsRequestBuilder implements RequestBuilder<ApiVersionsRequest> {
+  private static readonly apiKey = 18;
+
+  constructor(
+    private readonly clientId: NullableString,
```

Comment:
Because the exact types of the same field may vary between different versions of the requests (which is not the case in the ApiVersions request), the request builder should accept standard types (i.e. `string | null`) and do the translation (i.e. to the `NullableString`) internally.